src/utils/errorSerializer.ts

```diff
@@ -34,60 +38,81 @@ import UserTimeoutError from "../errors/UserTimeoutError";
 import VersionInvalidError from "../errors/VersionInvalidError";
 import VersionMismatchError from "../errors/VersionMismatchError";
 
-const errorCodeToErrorClass: {[key: string]: any} = {
+const errorCodeToErrorClass: Record<keyof typeof ErrorCode, KnownErrorConstructors> = {
   [ErrorCode.ERR_WEBEID_ACTION_PENDING]:          ActionPendingError,
   [ErrorCode.ERR_WEBEID_ACTION_TIMEOUT]:          ActionTimeoutError,
   [ErrorCode.ERR_WEBEID_CONTEXT_INSECURE]:        ContextInsecureError,
   [ErrorCode.ERR_WEBEID_EXTENSION_UNAVAILABLE]:   ExtensionUnavailableError,
+  [ErrorCode.ERR_WEBEID_MISSING_PARAMETER]:       MissingParameterError,
   [ErrorCode.ERR_WEBEID_NATIVE_INVALID_ARGUMENT]: NativeInvalidArgumentError,
   [ErrorCode.ERR_WEBEID_NATIVE_FATAL]:            NativeFatalError,
   [ErrorCode.ERR_WEBEID_NATIVE_UNAVAILABLE]:      NativeUnavailableError,
   [ErrorCode.ERR_WEBEID_USER_CANCELLED]:          UserCancelledError,
   [ErrorCode.ERR_WEBEID_USER_TIMEOUT]:            UserTimeoutError,
   [ErrorCode.ERR_WEBEID_VERSION_INVALID]:         VersionInvalidError,
   [ErrorCode.ERR_WEBEID_VERSION_MISMATCH]:        VersionMismatchError,
+  [ErrorCode.ERR_WEBEID_UNKNOWN_ERROR]:           UnknownError,
 };
 
-export function serializeError(error: any): any {
-  const {
-    message,
-    name,
-    fileName,
-    lineNumber,
-    columnNumber,
-    stack,
-  } = error;
-
-  return {
-    ...(
-      Object.fromEntries(
-        Object.getOwnPropertyNames(error)
-          .map((prop) => [prop, error[prop]])
-      )
-    ),
-
-    message,
-    name,
-    fileName,
-    lineNumber,
-    columnNumber,
-    stack,
-  };
+export function serializeError(error: Error): SerializedError {
+  if (typeof error != "object" || !(error instanceof Error)) {
+    error = new UnknownError(typeof error == "string" ? error : undefined);
+  }
+
+  const result: Record<string, unknown> = {};
+
+  for (const prop of Object.getOwnPropertyNames(error)) {
+    Reflect.set(result, prop, Reflect.get(error, prop));
```

Comment:
Wouldn't direct assignment be more readable:
```suggestion
    result[prop] = error[prop];
```
Or is there any specific reason to use `Reflect`?

Reply:
Here `prop` is an unknown string value and can't be used to reference `Error` object's properties - a down side of having strict linter rules.

This would satisfy the linter:
```ts
result[prop] = error[prop as 'name' | 'stack' | 'message'];
```
But this would be misleading as the error object would have other properties and some of them depend on the browser.

Another issue is that if the `error` object is a KnownError, it would have the `code` property which we also want to copy to the serialized result. The following wouldn't work tho:
```ts
result[prop] = error[prop as 'name' | 'stack' | 'message' | 'code'];
```
That's because the `Error` object doesn't have the `code` property.  

Using Reflect, when the property is dynamic, is probably the cleanest way to do it, unless we change the linter rules.

In the following case, the use of reflect isn't actually needed. This is because the `result` is of `SerializedError` type and we can access `result.code` and `result.name` directly.
```ts
  if (!result.code) {
    Reflect.set(result, "code", ErrorCode.ERR_WEBEID_UNKNOWN_ERROR);
    Reflect.set(result, "name", UnknownError.name);
  }
```
I'll change that.